src/plugins/intel_gpu/include/intel_gpu/runtime/lru_cache.hpp

```diff
@@ -164,6 +179,49 @@ class LruCache {
     }
 };
 
-using ImplementationsCache = cldnn::LruCache<size_t, std::shared_ptr<primitive_impl>>;
 using KernelsCache = cldnn::LruCache<size_t, cldnn::kernel::ptr>;
+
+template<typename Key, typename Value>
+class LruCacheThreadSafe {
```

Comment:
Can't we just make it to inherit LruCache class ? 
Seems that we are creating two objects by this change.

Reply:
Done.